### PR TITLE
Support Scala 3.6 in TASTy reader

### PR DESCRIPTION
**Pull request by SethTisue**

### CODE REVIEW

**Comment by SethTisue** [comment]
```
TestArrayAnnot.scala:5: error: exception during macro expansion:
java.lang.AssertionError: assertion failed: actually, was new SuppressWarnings(value = Array.type.apply[String]((Array[String]{"xyz", "foo"}: String*))(reflect#ClassTag.type.apply[String](classOf[java.lang.String])))
	at scala.Predef$.assert(Predef.scala:279)
	at tastytest.package$Macros$AnnotsBundle.forceAnnotsImpl(package.scala:57)

    forceAnnots[
               ^
```

**Comment by SethTisue** [comment]
@bishabosha (if you're available) what do you make of the failure?

File: project/DottySupport.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ import sbt.librarymanagement.{
   * Settings to support validation of TastyUnpickler against the release of dotty with the matching TASTy version
   */
 object TastySupport {
-  val supportedTASTyRelease = "3.5.2-RC2" // TASTY: 28.5-1
+  val supportedTASTyRelease = "3.6.1" // TASTY: 28.5-0
```
Comment (bishabosha) [inline project/DottySupport.scala]:
```suggestion
  val supportedTASTyRelease = "3.6.1" // TASTY: 28.6-0
```

**Comment by SethTisue** [comment]
might the root cause be https://github.com/scala/scala3/pull/21329?

**Comment by bishabosha** [comment]
firstly the `TastyFormat.scala` file should be copy-pasted (Except package name) from dotty at the 3.6.1 tag (which will update the TASTy version to 28.6.0)

not sure about this error until that is done

File: project/DottySupport.scala (hunk 1 above)
Comment (SethTisue) [reply project/DottySupport.scala]: thanks. I see now that this is what tasty/src/dotty/tools/tasty/TastyFormat.scala has at the 3.6.1 tag — I must have somehow done it wrong when I looked earlier

**Comment by SethTisue** [comment]
I'm seeing the same error locally even after updating TastyFormat.scala

**Comment by bishabosha** [comment]
@SethTisue the problem is a String comparison error, it expects the string `"new SuppressWarnings(Array.type.apply[String]((Array[String]{\"xyz\", \"foo\"}: String*))(reflect#ClassTag.type.apply[String](classOf[java.lang.String])))"` but as the error shows it now has a named argument

it should be fine to update the string in test/tasty/pos/src-2/tastytest/TestArrayAnnot.scala

**Comment by bishabosha** [comment]
seems also that the changes to context bound desugaring to using clauses means the signature of EnumCompanion also changed.

Anyway I pushed the fix to the test sources

**Pull request merged**